### ISSUE TITLE
Passing list of values as query parameter is broken in uplink 0.6.0

### ISSUE THREAD

**Issue by daa** [prkumar/uplink#108]
Consider we have some consumer with defined query parameter:
```
class C(uplink.Consumer):
    @uplink.args(uplink.Query())
    @uplink.get('/')
    def root(self, p):
        pass
```

And if we call ``c.root(p=['p1', 'p2'])`` ``uplink`` transforms string parameters to list, then that list to string . Thus ``uplink`` makes request like following:
```
GET /?p=%5B%27p%27%2C+%271%27%5D&p=%5B%27p%27%2C+%272%27%5D HTTP/1.1
```
and after decoding query parameters we'll see ``"['p', '1']"`` as the first parameter and ``"['p', '2']"`` as the second but expected values are ``"p1"`` and ``"p2"``.

However if we specify query type to ``str``: ``uplink.Query(type=str)`` such conversion does not happen and uplink makes correct request. Also specifying ``type=lambda x: x`` helps too. This is workaround I'll use for a while.

I found that triggering change was try to use value type to get converter:
```
-        argument_type, converter_key = self.type, self.converter_key
+        argument_type = self.type or type(value)
```

From investigating the code I found that the type of argument is used to find converter for all argument's parts if it is composite. In our case argument is list of strings ``['p1', 'p2']``, uplink uses key ``Sequence(CONVERT_TO_STRING)`` to find converter for query parameters, and then type ``list`` is used to find converter for ``'p1'`` and ``'p2'`` and corresponding ``Cast(list, StringConverter(...))`` is found which is incorrect and mangles parameters.

**Comment by prkumar** [comment]
@daa - Thanks for opening the issue and for the detailed investigation! I think the most straightforward option here is to revert the triggering change; after some thought, I think the new behavior makes conversion less explicit, whereas the conversion behavior should be opt-in, as it was before.

**Comment by daa** [comment]
Thank you for explanation. I agree that reverting triggering change is the easiest way to deal with it.

**Comment by prkumar** [comment]
I've merged the fix into `master` and `stable`, and the change should go live with the v0.6.1 hot-fix by tomorrow night (PDT).

**Comment by prkumar** [comment]
The fix went live with v0.6.1 release just now. 

FYI: @daa
